src/compiler/binder.ts

```diff
@@ -224,12 +224,24 @@ namespace ts {
 
                     // Report errors every position with duplicate declaration
                     // Report errors on previous encountered declarations
-                    let message = symbol.flags & SymbolFlags.BlockScopedVariable
+
+                    // @todo(marty) Either add a type or use the ternary operator.
+                    let message: any;
+
+                    if (symbol.name === "default") {

```

Comment:
After applying other changes I've suggested, separately test if you can test `declaration.flags & NodeFlags.Default`.